### PR TITLE
PSMDB-139 Improve compaction implementation

### PR DESCRIPTION
**Pull request by denis-protivenskii**
1. Compaction-related code is grouped into RocksCompactionScheduler.
2. Add compaction of dropped prefixes markers.
3. Prioritize different types of compactions.

### CODE REVIEW

File: src/rocks_util.h [hunk 1]
```diff
@@ -48,6 +48,10 @@ namespace mongo {
         return nextPrefix;
     }
 
+    std::string encodePrefix(uint32_t prefix);
+    bool extractPrefix(const rocksdb::Slice& slice, uint32_t* prefix);
+    int get_internal_delete_skipped_count();
```
Comment (igorcanadi) [inline src/rocks_util.h]: These are small enough that we can make them `inline`, but no need to do that now, only if you feel like it in the next PR.

File: src/rocks_compaction_scheduler.cpp [hunk 2]
```diff
@@ -75,26 +153,34 @@ namespace mongo {
...
     }
 
     CompactionBackgroundJob::~CompactionBackgroundJob() {
         {
             stdx::lock_guard<stdx::mutex> lk(_compactionMutex);
             _compactionThreadRunning = false;
-            _compactionQueue.clear();
+            // Clean up the queue
+            CompactQueue tmp;
+            _compactionQueue.swap(tmp);
```
Comment (igorcanadi) [inline src/rocks_compaction_scheduler.cpp]: Interesting, is that faster than `clear`?

Comment (denis-protivenskii) [reply src/rocks_compaction_scheduler.cpp]: It's just because `std::priority_queue` doesn't have `clear()` method.

**Comment by denis-protivenskii** [comment]
Resolved conflicts, it's now ready to be merged.

File: src/rocks_util.h (hunk 1 above)
Comment (denis-protivenskii) [reply src/rocks_util.h]: I'll do it in the next PR, np.

**Pull request merged**